internal/server/handlers_test.go

```diff
@@ -710,7 +710,7 @@ func TestAPI_ListGrantsV0_12_2(t *testing.T) {
 	routes := srv.GenerateRoutes(prometheus.NewRegistry())
 
 	resp := httptest.NewRecorder()
-	req, err := http.NewRequest(http.MethodGet, "/v1/grants?privilege=admin", nil)
+	req, err := http.NewRequest(http.MethodGet, "/v1/grants?privilege=%20admin", nil)
```

Comment:
Test coverage for this is great! To be a bit more explicit about what is being tested, maybe we could have a separate test case that includes whitespace (newline, tab, space, vertical space) in both query parameters and request body? 

That test case would also probably be a great place to include some negative tests, to show that a `[]byte` with leading or trailing whitespace is preserved.